src/Common/src/Interop/Windows/NtDll/Interop.NtCreateFile.cs

```diff
@@ -33,7 +36,23 @@ internal partial class NtDll
             System.IO.FileShare shareAccess = System.IO.FileShare.ReadWrite | System.IO.FileShare.Delete,
             System.IO.FileAttributes fileAttributes = 0,
             CreateOptions createOptions = CreateOptions.FILE_SYNCHRONOUS_IO_NONALERT,
-            ObjectAttributes objectAttributes = ObjectAttributes.OBJ_CASE_INSENSITIVE)
+            ObjectAttributes objectAttributes = ObjectAttributes.OBJ_CASE_INSENSITIVE
+            )
+        {
+                return ( CreateFile(eaBuffer: new Microsoft.Win32.SafeHandles.SafeFileHandle(IntPtr.Zero, true), eaLength: 0, path, rootDirectory,createDisposition, desiredAccess, shareAccess, fileAttributes, createOptions, objectAttributes) );
```

Comment:
Don't do this- you're adding an allocation for every directory to all file enumerations on Windows. Switch back to null and pointer- I'll give a comment on how to deal with your new overload.